src/components/Doctors/MapEvents.js

```diff
@@ -16,6 +19,23 @@ const MapEvents = function MapEvents() {
         searchValue,
       });
       setDoctors(mapDoctors);
+      const newZoom = map.getZoom();
+      const newCenter = map.getCenter();
+
+      const oldHash = location.hash.replace('#', '').split('|');
+
+      const oldHashValues = oldHash?.reduce(reduceHash, {
+        search: null,
+        loc: null,
+        zoom: null,
+        accepts: null,
+      });
+
+      const hash = `a-${oldHashValues.accepts}|l-${newZoom}/${[newCenter.lat, newCenter.lng].join(
+        '/',
+      )}|s-${oldHashValues.search}`;
+
+      navigate(`./#${hash}`);
```

Comment:
It should not push a new entry into the browser history - pressing back button now only changes search parameters in the url, and possibly moving the map view. It should go to the previous page.

Maybe something like:
```suggestion
      navigate(`./#${hash}`, { replace: true });
```
? Didn't test that.